src/controllers/queries/index.ts

```diff
@@ -246,32 +312,22 @@ export class QueriesController extends HTMLController {
       companyId: UUID
       action: 'success'
       totalScope3CarbonEmissions: string
-      partialResponse?: number
     }
   ): Promise<HTML> {
     this.logger.debug('query page requested')
 
-    const [connection]: ConnectionRow[] = await this.db.get(
-      'connection',
-      { id: body.companyId, status: 'verified_both' },
-      [['updated_at', 'desc']]
-    )
+    const [connection] = await this.db.get('connection', { id: body.companyId, status: 'verified_both' }, [
+      ['updated_at', 'desc'],
+    ])
     if (!connection) {
-      throw new InvalidInputError(`Invalid connection ${body.companyId}`)
+      throw new NotFoundError(`Invalid connection ${body.companyId}`)
     }
     if (!connection.agent_connection_id || connection.status !== 'verified_both') {
       throw new InvalidInputError(`Cannot query unverified connection`)
     }
-
-    const [queryRow]: QueryRow[] = await this.db.get('query', { id: queryId })
-    if (!queryRow) {
-      throw new NotFoundError(`There has been an issue retrieving the query.`)
-    } else if (!queryRow.response_id) {
-      throw new InvalidInputError('Missing queryId to respond to.')
-    } else if (queryRow.connection_id !== connection.id) {
-      throw new NotFoundError(`Missing queryId to respond to.`)
-    } else if (queryRow.status === 'errored') {
-      throw new InvalidInputError('Cannot process query with status - "errored"')
+    const [queryRow] = await this.db.get('query', { id: queryId })
+    if (!queryRow.response_id) {
```

Comment:
`queryRow` could be falsey and if so should throw `NotFoundError`. 

We should probably also be making sure that the `query` is for the company provided. A better alternative would also be to not provide the company in the body and to instead look it up from the `query`

Reply:
going to update this right after the food

Reply:
has this point been addressed? : 
`queryRow` could be falsey and if so should throw `NotFoundError`

Reply:
it has been updated